### ISSUE TITLE
Queue destroy callback

### ISSUE THREAD

**Issue by jacobstr** [postwait/node-amqp#226]
It would be nice to have a callback for the successful/failed destruction of a queue.

E.g. we have a long running process that orchestrates a multi-step job. Once this is done, I'd like to clean the queues up. Currently, I iterate over each queue and destroy it, then exit after a timeout period but it's slightly unreliable. From my experiments, if the connection is closed before the queues have been destroyed they will not be destroyed.
